src/Raven.Client/Json/Serialization/NewtonsoftJson/Internal/BlittableJsonWriter.cs

```diff
@@ -25,6 +30,10 @@ internal class BlittableJsonWriter : JsonWriter, IJsonWriter
             _manualBlittableJsonDocumentBuilder.StartWriteObjectDocument();
             _documentInfo = documentInfo;
             _first = true;
+
+            _metadataKey = context.GetLazyStringForFieldWithCaching(Constants.Documents.Metadata.Key);
+            _metadataCollection = context.GetLazyStringForFieldWithCaching(Constants.Documents.Metadata.Collection);
+            _metadataId = context.GetLazyStringForFieldWithCaching(Constants.Documents.Metadata.Id);
```

Comment:
Shouldn't we dispose them on writer dispose?

Reply:
No, note that this is used with caching, so it will live for the life time of the context.